### DISCUSSION TITLE
move asdf-scala here

### DISCUSSION THREAD

**Discussion by cj81499** [asdf-community/.github#20] [General]
It seems the maintainer of [asdf-scala](https://github.com/mtatheonly/asdf-scala) has been unresponsive for quite some time, and the plugin is broken for Scala 3+.

I'm not sure if there's any precedent/procedure, but hopefully we can work something out!

**Comment by cj81499** [comment]
In the meantime, if anyone wants scala to work with scala3+, try the following:
```shell
# remove the scala plugin (if you already have it)
asdf plugin remove scala
# install an asdf-scala fork with scala3 support. thanks @paulmr!
asdf plugin add scala git@github.com:paulmr/asdf-scala.git
# switch to the plugin to the scala3 branch
git -C $ASDF_DIR/plugins/scala/ checkout scala3
# profit (install the latest scala, 3.0.1 at time of writing)
asdf install scala latest
```

**Comment by smorimoto** [comment]
ping @mtatheonly

**Comment by smorimoto** [comment]
@vic You seem to be more familiar with the community than I am - was there any other plugin for Scala? Is it still necessary in the first place? (I would like to see if there is an official version manager...)

**Reply by vic** [reply]
Dunno, never used scala directly from an asdf-plugin :D. I guess it's only useful if you want to run scalac/scala via the command line.

I also don't know if there's another plugin, I did a dotty one but it was prior dotty becoming Scala3, but again it was for playing around with dotty repl via the command line. sbt and mill both have had support for scala3 projects since way a lot of time.

I'm no longer using asdf, but my co-workers still use [asdf-mill](https://github.com/asdf-community/asdf-mill) for their setup.

For most professional scala projects, you mostly use a build tool (main one is sbt but I prefer mill a lot more) and you configure the build tool and set your project's scala version(s) - it can be many versions since you might want to cross-compile/run your project on many scala versions (eg, 2.x and 3.x, scala.js or scala-native). And the build tool will download the compiler and runtime-libs for such versions via [coursier](https://get-coursier.io) which has become a kind-of defacto scala/jvm/libs dependency fetcher.

If I was not using Nix now, I'd prefer to have `coursier` installed on PATH and run `coursier launch --scala 3.1.0 scala` or the like whenever trying to use some tooling from the command line. But that's just me. If people feel like having a `scala`/`scalac` command on their PATH an asdf-plugin would be useful.

**Reply by smorimoto** [reply]
Oh I see I see, I fully understood them now! Thank you @vic!
(It's a bit off the main line, but I'm excited to see that the Scala community is getting a lot better than I can remember!)

**Comment by mtatheonly** [comment]
hey guys, I can move repository to asdf-community group. what's needed to be done?

**Reply by smorimoto** [reply]
Cool! I will send the invitation to your account! Then, you can just transfer the repository here...

**Reply by smorimoto** [reply]
Done by https://github.com/asdf-community/infrastructure/commit/98e941dfa059ecc4929a440482f7b88e800418c3.

**Reply by mtatheonly** [reply]
repository moved =)

**Reply by smorimoto** [reply]
Thank you!
